src/main/java/com/maddyhome/idea/vim/extension/replacewithregister/ReplaceWithRegister.kt

```diff
@@ -148,6 +149,22 @@ private fun doReplace(editor: Editor, context: DataContext, caret: ImmutableVimC
 
   var usedType = savedRegister.type
   var usedText = savedRegister.text
+
+  val selection = visualSelection.caretsAndSelections.values.first()
+  val startOffset: Int = selection.vimStart
+  val endOffset: Int = selection.vimEnd + 1
+
+  val isEmptySelection: Boolean = startOffset == endOffset
+
+  if (isEmptySelection) {
+    val editor: VimEditor = editor.vim
+
+    val beforeChar: String = editor.getText(startOffset - 1, startOffset)
+    val afterChar: String = editor.getText(endOffset, endOffset + 1)
+
+    usedText = beforeChar + usedText + afterChar
```

Comment:
Well, this might give you the correct execution results, but the approach is fundamentally incorrect. Instead of inserting the string between, say, brackets, you’re actually removing the brackets and inserting the text with additional wrapping.
While it may seem to work for now, it could lead to issues during future refactorings, since the operation being performed differs from what is expected.